Core/clim-basic/input.lisp

```diff
@@ -120,10 +120,11 @@ is no such event."))
   (:documentation "Returns true if there are any events in the queue. Otherwise
 returns NIL."))
 
-(defgeneric event-queue-listen-or-wait (event-queue &key timeout)
-  (:documentation "Returns true if there are any events in the queue. Otherwise
-blocks until event arives or timeout is reached. If there is no event before
-timeout returns NIL."))
+(defgeneric event-queue-listen-or-wait (event-queue &key timeout wait-function)
```

Comment:
pending events won't be processed, only appended to the queue. In principle it should return immedietely when wait-function returns t and events are considered being a source of state changes. In `concurrent-event-queue` specification we even clamp timeout to 0.01 to assure that asynchronous state changes (albeit undesired) will lead to end of waiting.

Reply:
I see. But just from reading the documentation it isn't clear why one would want to call it. The name of the function gives some hint, but it would be nicer if the docs were more clear.

Reply:
how would you rephrase it then? it is called without arguments

Reply:
if wait-function returns NIL then event-queue-listen-or-wait does not return. function returns T if wait-function is T and returns NIL if timeout is met. there is no third option.